src/include/btree.i

```diff
@@ -1310,20 +1310,40 @@ __wt_page_swap_func(WT_SESSION_IMPL *session, WT_REF *held,
 #endif
 	    );
 
-	/* Expected failures: page not found or restart. */
-	if (ret == WT_NOTFOUND || ret == WT_RESTART)
-		return (ret);
+	/*
+	 * Expected failures: page not found or restart. Our callers list the
+	 * errors they're expecting to handle.
```

Comment:
Actually, to me, this makes my earlier suggestion of using `WT_READ_RESTART_OK` even more readable because it implies the error is coming from the read/page-in function.